src/useTracking.js

```diff
@@ -0,0 +1,29 @@
+import merge from 'deepmerge';
+import { useContext, useCallback } from 'react';
+import { ReactTrackingContext } from './withTrackingComponentDecorator';
+
+export default function useTracking() {
+  const trackingContext = useContext(ReactTrackingContext);
+
+  if (!(trackingContext && trackingContext.tracking)) {
+    throw new Error(
+      'Attempting to call `useTracking` ' +
+        'without a ReactTrackingContext present. Did you forget to wrap the top of ' +
+        'your component tree with `track`?'
+    );
+  }
+
+  const trackEvent = useCallback(
+    data => {
+      return trackingContext.tracking.dispatch(
+        merge(trackingContext.tracking.data, data)
+      );
+    },
+    [trackingContext.tracking.data]
+  );
+
+  return {
+    getTrackingData: () => trackingContext.tracking.data,
+    trackEvent,
+  };
```

Comment:
Ah nice idea to `useCallback` @damassi  , but I think the issue though is that this returns a new object every time. Would something like the following make sense?

```js
return useMemo(() => ({
  getTrackingData, () => trackingContext.tracking.data,
  trackEvent: data => trackingContext.tracking.dispatch(merge(trackingContext.tracking.data, data))
}), [trackingContext.tracking.data])
```

Reply:
Yup! will update.